### ISSUE TITLE
Rule proposal: prefer-object-has-own

### ISSUE THREAD

**Issue by axetroy** [sindresorhus/eslint-plugin-unicorn#2479]
### Description

Prefer `Object.hasOwn()` over `Object.hasOwnProperty`

`Object.hasOwnProperty()` actually has a trap.

If the call method is not called, it will return false regardless.

`Object.hasOwn()` is more safe

### Fail

```js
Object.hasOwnProperty(data, 'onclick') // Wrong way of using
Object.prototype.hasOwnProperty(data, 'onclick') // Wrong way of using
Object.prototype.hasOwnProperty.call(data, "onclick")
```

### Pass

```js
Object.hasOwn(data, "onclick")
Object.hasOwnProperty(data) // ignore when not called with 2 arguments
Object.hasOwnProperty(data, "onclick", 3) // ignore when not called with 2 arguments
Object.prototype.hasOwnProperty.call(data, "onclick", 3) // ignore when not called with 2 arguments
```

### Proposed rule name

prefer-object-has-own

### Additional Info

```
exports[`prefer-object-has-own 1`] = `
## invalid(1): Object.hasOwnProperty(data, 'onclick')

  > Input
  
        1 | Object.hasOwnProperty(data, 'onclick')

  > Error 1/1: Using Object.hasOwnProperty() like this is the wrong way to do it
  
      > 1 | Object.hasOwnProperty(data, 'onclick')
          | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  > Suggestion 1/1: Use Object.hasOwn() instead
  
        1 | Object.hasOwn(data, 'onclick')

## invalid(2): Object.prototype.hasOwnProperty(data, 'onclick')

  > Input
  
        1 | Object.prototype.hasOwnProperty(data, 'onclick')

  > Error 1/1: Using Object.hasOwnProperty() like this is the wrong way to do it
  
      > 1 | Object.prototype.hasOwnProperty(data, 'onclick')
          | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  > Suggestion 1/1: Use Object.hasOwn() instead
  
        1 | Object.hasOwn(data, 'onclick')

## invalid(3): Object.hasOwnProperty.call(data, "onclick")

  > Input
  
        1 | Object.hasOwnProperty.call(data, "onclick")

  > Output
  
        1 | Object.hasOwn(data, "onclick")

  > Error 1/1: Use Object.hasOwn() instead
  
      > 1 | Object.hasOwnProperty.call(data, "onclick")
          | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

## invalid(4): Object.prototype.hasOwnProperty.call(data, "onclick")

  > Input
  
        1 | Object.prototype.hasOwnProperty.call(data, "onclick")

  > Output
  
        1 | Object.hasOwn(data, "onclick")

  > Error 1/1: Use Object.hasOwn() instead
  
      > 1 | Object.prototype.hasOwnProperty.call(data, "onclick")
          | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

## invalid(5): Object\\n.hasOwnProperty(data, 'onclick')

  > Input
  
        1 | Object
        2 | .hasOwnProperty(data, 'onclick')

  > Error 1/1: Using Object.hasOwnProperty() like this is the wrong way to do it
  
      > 1 | Object
          | ^^^^^^
      > 2 | .hasOwnProperty(data, 'onclick')
          | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  > Suggestion 1/1: Use Object.hasOwn() instead
  
        1 | Object
        2 | .hasOwn(data, 'onclick')
`;

```

**Comment by fisker** [comment]
https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1314

**Comment by axetroy** [comment]
In fact, there are overlaps with the official rules, but they are not exactly the same.

The official rule does not check `Object.hasOwnProperty(data, 'onclick')`, which is a wrong usage

**Comment by fisker** [comment]
Not sure what you mean, but you can try to raise an issue in the eslint repo.
